Common/src/Pollster/TaskHandler.cs

```diff
@@ -490,36 +472,47 @@ await taskTable_.RetryTask(taskData_,
-            finally
-            {
-              var retryData = await taskTable_.ReadTaskAsync(retryId,
-                                                             lateCts_.Token)
-                                              .ConfigureAwait(false);
-              if (retryData.Status is TaskStatus.Creating or TaskStatus.Submitted)
-              {
-                logger_.LogWarning("Retried task {task} is in {status}; trying to finalize task creation",
-                                   retryId,
-                                   retryData.Status);
-                await submitter_.FinalizeTaskCreation(new List<TaskCreationRequest>
-                                                      {
-                                                        new(retryId,
-                                                            retryData.PayloadId,
-                                                            retryData.Options,
-                                                            retryData.ExpectedOutputIds,
-                                                            retryData.DataDependencies),
-                                                      },
-                                                      sessionData_,
-                                                      taskData_.TaskId,
-                                                      CancellationToken.None)
-                                .ConfigureAwait(false);
-              }
-            }
 
-            return AcquisitionStatus.TaskIsRetriedAndRetryIsNotFound;
+            retryData = await taskTable_.ReadTaskAsync(retryId,
+                                                       CancellationToken.None)
+                                        .ConfigureAwait(false);
+          }
+
+          if (retryData.Status is TaskStatus.Creating or TaskStatus.Submitted)
+          {
+            logger_.LogWarning("Retried task {task} is in {status}; trying to finalize task creation",
+                               retryId,
+                               retryData.Status);
+            await submitter_.FinalizeTaskCreation(new List<TaskCreationRequest>
+                                                  {
+                                                    new(retryId,
+                                                        retryData.PayloadId,
+                                                        retryData.Options,
+                                                        retryData.ExpectedOutputIds,
+                                                        retryData.DataDependencies),
+                                                  },
+                                                  sessionData_,
+                                                  taskData_.TaskId,
+                                                  CancellationToken.None)
+                            .ConfigureAwait(false);
+          }
+
+          switch ((taskNotFound, taskExists, retryData.Status))
```

Comment:
I think switch expression would be simpler, and I think you can select on multiple patterns with `or`